### PR TITLE
009-Course API

### PR DESCRIPTION
**Pull request by ellamaolson**
A schema for 'course' that contains related information about courses as well as basic API such as creating, fetching, editting, delete, etc. so that I can build complex functionalities using that schema.

### CODE REVIEW

File: server/controller/course.controller.js [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+          termsOffered
+        } = req.body;
+
+        const course = await Course.findOne({
+          school,
+          code
+        });
+        if (!course) {
+          const createdCourse = await createCourse(school, code, title, description, prerequisites, corequisites, difficulty, impaction, termsOffered);
+          const courseToReturn = {
```
Comment (DaleCS) [inline server/controller/course.controller.js]: No need to parse the object into JSON. The json() function at the end of your response statement already does that for you.

File: server/controller/course.controller.js [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+  .withMessage(DOES_NOT_CONTAIN_A_DIGIT)
+];
+
+const validateSchoolCourses = [
+  check('school')
+  .exists()
+  .withMessage(COURSE_SCHOOL_IS_EMPTY),
+];
+
+function createCourse(school, code, title, description, prerequisites, corequisites, difficulty, impaction, termsOffered) {
```
Comment (calvinqc) [inline server/controller/course.controller.js]:
```
function createCourse(data) {
...
}
```

File: server/controller/course.controller.js [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+          impaction,
+          termsOffered
+        } = req.body;
+
+        const course = await Course.findOne({
+          school,
+          code
+        });
+        if (!course) {
+          const createdCourse = await createCourse(school, code, title, description, prerequisites, corequisites, difficulty, impaction, termsOffered);
```
Comment (calvinqc) [inline server/controller/course.controller.js]: `const createdCourse = await createCourse({ school, code, title, description, prerequisites, corequisites, difficulty, impaction, termsOffered });`

This way, you don't have to define `const data` in createCourse.

File: server/controller/course.controller.js [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+
+  course = new Course(data);
+  return course.save();
+}
+
+/**
+ * POST/
+ * Create a new course, requires authentication of user
+ */
+courseController.post('/create',
```
Comment (calvinqc) [inline server/controller/course.controller.js]: usually, the standard for POST to create an object doesn't need to have `/create`, we can just use '/' -> Developers will know it's POST/ for create an object.

File: server/database/schemas/course.schema.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import { Schema } from 'mongoose';
+
+const courseSchema = new Schema({
+  school: {type: String, required: true, uppercase: true },
+  code: { type: String, required: true, uppercase: true },
+  title: { type: String, required: true },
+  description: { type: String, required: true },
+  prerequisites: { type: [Object], uppercase: true  },
```
Comment (calvinqc) [inline server/database/schemas/course.schema.js]: More Specifically:
`prerequisites: [{ type: Schema.Types.ObjectId, ref: 'Course' , required: true, uppercase: true }],
 corequisites: [{ type: Schema.Types.ObjectId, ref: 'Course' , required: true, uppercase: true }]`

**Review by DaleCS — APPROVED** [review]
It do be look good to me.

**Pull request merged**